api/v1alpha3/conversion.go

```diff
@@ -388,3 +389,11 @@ func RestoreRootVolume(restored, dst *v1alpha4.Volume) {
 	dst.Throughput = restored.Throughput
 	return
 }
+
+// Convert_v1alpha4_AWSLoadBalancerSpec_To_v1alpha3_AWSLoadBalancerSpec is based on the autogenerated function and handles the Name field.
+func Convert_v1alpha4_AWSLoadBalancerSpec_To_v1alpha3_AWSLoadBalancerSpec(in *v1alpha4.AWSLoadBalancerSpec, out *AWSLoadBalancerSpec, s apiconversion.Scope) error {
+	if in.Name != nil && *in.Name != "" {
```

Comment:
As the load balancer `Name` is a new field in v1alpha4, it doesn't need to be converted down to v1alpha3, meaning if statement here is not needed.

It should be restored from the annotation when converting back to v1alpha4 in [ConvertTo](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/api/v1alpha3/conversion.go#L32) function to pass the round trip tests.

Reply:
Thanks @pydctw.

When new a field is introduced, the field can be lost in down-conversion, as raised in https://github.com/kubernetes-sigs/cluster-api/issues/1953. Cluster API's solution to avoid losing information during down-conversion is to place fields in a special annotation, first introduced in https://github.com/kubernetes-sigs/cluster-api/pull/1992.

I hadn't noticed this before.

Reply:
Fixed. Thanks again!